### PR TITLE
[Accordion] Support multiple values

### PR DESCRIPTION
**Pull request by jjenzz**
- Adds support for multiple values
- Toggles items by default (clicking button again will close item)
- To prevent items from closing, consumer now needs to control it
- Updates tests
- Updates stories

### CODE REVIEW

File: packages/react/accordion/src/Accordion.tsx [hunk 1]
```diff
@@ -37,46 +67,136 @@ type AccordionOwnProps = Polymorphic.Merge<
...
 
+  const handleItemOpen = React.useCallback(
+    (itemValue) => setValue((prevValue = []) => [...prevValue, itemValue]),
+    [setValue]
+  );
+
+  const handleItemClose = React.useCallback(
+    (itemValue) => setValue((prevValue = []) => prevValue.filter((value) => value !== itemValue)),
+    [setValue]
+  );
```
Comment (jjenzz) [inline packages/react/accordion/src/Accordion.tsx:157]: This is a good example of why it's nice to have the callback implementation in the component that provides the context. I have the same context API but two different handler implementations 👍

File: packages/react/accordion/src/Accordion.tsx [hunk 2]
```diff
@@ -37,46 +67,136 @@ type AccordionOwnProps = Polymorphic.Merge<
...
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  AccordionImplOwnProps
+>;
+
+const AccordionImpl = React.forwardRef((props, forwardedRef) => {
+  const { disabled, ...accordionProps } = props;
+  const buttonNodesRef = React.useRef<Set<React.ElementRef<typeof AccordionButton>>>(new Set());
+  const accordionRef = React.useRef<React.ElementRef<typeof AccordionImpl>>(null);
+  const composedRefs = useComposedRefs(accordionRef, forwardedRef);
+
```
Comment (jjenzz) [inline packages/react/accordion/src/Accordion.tsx:199]: These aren't new https://github.com/radix-ui/primitives/pull/527/files#diff-4da10955329003056dec9c69a6f4f8b1fe4542d50f1747908130d6bc50a22cbaL70-L73

File: packages/react/accordion/src/Accordion.tsx [hunk 3]
```diff
@@ -85,7 +205,7 @@ const Accordion = React.forwardRef((props, forwardedRef) => {
       return;
     }
 
-    const buttonNodes = [...buttonNodesRef.current].filter((node) => !(node && node.disabled));
+    const buttonNodes = [...buttonNodesRef.current].filter((node) => !node?.disabled);
```
Comment (jjenzz) [inline packages/react/accordion/src/Accordion.tsx:208]: Slipped this change in, forgive me 🙈

File: packages/react/accordion/src/Accordion.stories.tsx [hunk 4]
```diff
@@ -11,74 +11,235 @@ import {
...
+          </AccordionHeader>
+          <AccordionPanel className={panelClass}>
+            Odio placerat <a href="#">quisque</a> sapien sagittis non sociis ligula penatibus
+            dignissim vitae, enim vulputate nullam semper potenti etiam volutpat libero.
+            <button>Cool</button>
+          </AccordionPanel>
+        </AccordionItem>
+      </Accordion>
+
+      <h1>Controlled (without close)</h1>
```
Comment (benoitgrelard) [inline packages/react/accordion/src/Accordion.stories.tsx:104]: I got confused by this title as I was expected none to close, but I guess this is maybe more like 

```suggestion
      <h1>Controlled (at least one open)</h1>
```

?

File: packages/react/accordion/src/Accordion.test.tsx [hunk 5]
```diff
@@ -60,44 +61,164 @@ describe('given a basic Accordion', () => {
...
+    let panelTwo: HTMLElement | null;
+
     beforeEach(async () => {
       button = rendered.getByText('Button One');
       fireEvent.click(button);
       await waitFor(() => {
-        panel = rendered.getByText('Panel One');
-        expect(panel).toBeVisible();
+        panelOne = rendered.getByText('Panel One');
+        return panelOne;
```
Comment (benoitgrelard) [inline packages/react/accordion/src/Accordion.test.tsx:73]: what does it do to return it here? Is that necessary considering it's not assigned?

File: packages/react/accordion/src/Accordion.test.tsx [hunk 6]
```diff
@@ -60,44 +61,164 @@ describe('given a basic Accordion', () => {
...
+    let button: HTMLElement;
+    let panelOne: HTMLElement | null;
+    let panelTwo: HTMLElement | null;
+
+    beforeEach(async () => {
+      button = rendered.getByText('Button One');
+      fireEvent.click(button);
+      await waitFor(() => {
+        panelOne = rendered.getByText('Panel One');
+        return panelOne;
```
Comment (benoitgrelard) [inline packages/react/accordion/src/Accordion.test.tsx:181]: same question

File: packages/react/accordion/src/Accordion.tsx [hunk 7]
```diff
@@ -37,46 +67,136 @@ type AccordionOwnProps = Polymorphic.Merge<
...
   const [value, setValue] = useControlledState({
     prop: valueProp,
     defaultProp: defaultValue,
-    onChange: (value) => value && onValueChange(value),
+    onChange: onValueChange,
+  });
+
+  return (
+    <AccordionValueProvider
+      value={value ? [value] : []}
```
Comment (benoitgrelard) [inline packages/react/accordion/src/Accordion.tsx:98]: Nice, so from the context perspective it's always an array?

File: packages/react/accordion/src/Accordion.tsx (hunk 1 above)
Comment (benoitgrelard) [reply packages/react/accordion/src/Accordion.tsx:157]: Love that

**Review by benoitgrelard — COMMENTED** [review]
This is great! Love the approach!
Left a few minor comments.

File: packages/react/accordion/src/Accordion.stories.tsx (hunk 4 above)
Comment (jjenzz) [reply packages/react/accordion/src/Accordion.stories.tsx:104]: Ah yes, that is exactly what it is, thanks :+1:

File: packages/react/accordion/src/Accordion.test.tsx (hunk 5 above)
Comment (jjenzz) [reply packages/react/accordion/src/Accordion.test.tsx:73]: I thought `waitFor` expects a promise returned to is knows what to wait for... could be wrong

File: packages/react/accordion/src/Accordion.tsx (hunk 7 above)
Comment (jjenzz) [reply packages/react/accordion/src/Accordion.tsx:98]: You got it :+1:

File: packages/react/accordion/src/Accordion.test.tsx (hunk 5 above)
Comment (jjenzz) [reply packages/react/accordion/src/Accordion.test.tsx:73]: Yeah, `return`s are completely redundant... removed!

**Review by benoitgrelard — APPROVED** [review]

**Pull request merged**